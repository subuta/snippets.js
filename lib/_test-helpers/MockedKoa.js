import { build, format, snippets as s } from 'bld.js'
import _ from 'lodash'

export default (srcDir = 'src') => {
  const imports = s.import([
    ['proxyquire', 'proxyquire'],
    ['../../config', null, [
      ['absolutePath']
    ]]
  ])

  return build`
    ${imports}
    
    // knex injected modules.
    
    export const model = (knex) => {
      return proxyquire(absolutePath('${srcDir}/model'), {
        '${srcDir}/utils/knex': knex
      }).default
    }
    
    export const api = (knex) => {
      const middleware = proxyquire(absolutePath('${srcDir}/api/middlewares/models'), {
        '${srcDir}/model': model(knex)
      }).default
    
      return proxyquire(absolutePath('${srcDir}/api'), {
        './middleware/models': middleware
      }).default
    }

  `
}
