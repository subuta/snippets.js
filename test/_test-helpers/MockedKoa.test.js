/* global expect, describe, it */

import { build, format, snippets as s } from 'bld.js'

import MockedKoa from 'lib/_test-helpers/MockedKoa'

describe('_test-helpers/MockedKoa', () => {
  it('should create mocked koa modules', () => {
    expect(format(MockedKoa())).toMatchSnapshot()
  })
})
