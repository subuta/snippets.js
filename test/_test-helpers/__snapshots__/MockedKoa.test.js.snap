// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_test-helpers/MockedKoa should create mocked koa modules 1`] = `
"import proxyquire from 'proxyquire'
import {absolutePath} from '../../config'

// knex injected modules.

export const model = (knex) => {
  return proxyquire(absolutePath('src/model'), {
    'src/utils/knex': knex
  }).default
}

export const api = (knex) => {
  const middleware = proxyquire(absolutePath('src/api/middlewares/models'), {
    'src/model': model(knex)
  }).default

  return proxyquire(absolutePath('src/api'), {
    './middleware/models': middleware
  }).default
}
"
`;
